monitoring/prober/scd/test_operation_simple_heavy_traffic.py

```diff
@@ -0,0 +1,302 @@
+    resp = scd_session.post('/operation_references/query', json={
+      'area_of_interest': req['extents'][0]
+    })
+    assert resp.status_code == 200, resp.content
+    found_ids = [op['id'] for op in resp.json().get('operation_references', [])]
+    ovns = [ovn_map[id] for id in found_ids]
+
+    # UPDATE operation
+    req = {
+      'key': ovns,
+      'extents': req['extents'],
+      'old_version': existing_op['version'],
+      'state': 'Activated',
+      'uss_base_url': 'https://example.com/uss2',
+      'subscription_id': existing_op['subscription_id']
+    }
+
+    resp = scd_session.put('/operation_references/{}'.format(op_id), json=req, scope=SCOPE_CI)
+    assert resp.status_code == 403, resp.content
+
+    resp = scd_session.put('/operation_references/{}'.format(op_id), json=req, scope=SCOPE_CM)
+    assert resp.status_code == 403, resp.content
+
+    resp = scd_session.put('/operation_references/{}'.format(op_id), json=req, scope=SCOPE_SC)
+    assert resp.status_code == 200, resp.content
+
+    data = resp.json()
+    op = data['operation_reference']
+    assert op['id'] == op_id
+    assert op['uss_base_url'] == 'https://example.com/uss2'
+    assert op['version'] == 2
+    assert op['subscription_id'] == existing_op['subscription_id']
+    assert 'state' not in op
+
+    ovn_map[op_id] = op['ovn']
+
+
+# Preconditions: Operations with ids in OP_IDS mutated to second version
+# Mutations: Operations with ids in OP_IDS deleted
+def test_delete_op(scd_session):
+  for op_id in OP_IDS:
+    resp = scd_session.delete('/operation_references/{}'.format(op_id), scope=SCOPE_CI)
+    assert resp.status_code == 403, resp.content
+
+    resp = scd_session.delete('/operation_references/{}'.format(op_id), scope=SCOPE_CM)
+    assert resp.status_code == 403, resp.content
+
+    resp = scd_session.delete('/operation_references/{}'.format(op_id), scope=SCOPE_SC)
+    assert resp.status_code == 200, resp.content
+
+
+# Preconditions: Operations with ids in OP_IDS deleted
+# Mutations: None
+@default_scope(SCOPE_SC)
+def test_get_deleted_ops_by_ids(scd_session):
+  for op_id in OP_IDS:
+    resp = scd_session.get('/operation_references/{}'.format(op_id))
+    assert resp.status_code == 404, resp.content
+
+
+# Preconditions: Operations with ids in OP_IDS deleted
+# Mutations: None
+@default_scope(SCOPE_SC)
+def test_get_deleted_ops_by_search(scd_session):
+  resp = scd_session.post('/operation_references/query', json={
+    'area_of_interest': common.make_vol4(None, None, 0, 5000, common.make_circle(-56, 178, 12000))
+  })
+  assert resp.status_code == 200, resp.content
+  found_ids = [op['id'] for op in resp.json().get('operation_references', [])]
+  assert not _intersection(OP_IDS, found_ids)
+
+
+# Generate request with volumes that cover a circle area that initially centered at (-56, 178)
+# The circle's center lat shifts 0.001 degree (111 meters) per sequential idx change
+def _make_op_request(idx):
```

Comment:
nit: Let's put utility functions at the top of the file so a reader has a chance to see them before seeing their use.

Reply:
Done, moved to the top of the file.